projects/ion/src/lib/badge/badge.component.ts

```diff
@@ -14,30 +12,16 @@ import { IonBadgeProps } from './types';
   templateUrl: './badge.component.html',
   styleUrls: ['./badge.component.scss'],
   host: {
-    '[attr.data-type]': 'type',
+    '[attr.data-type]': 'type()',
+    '[attr.data-testid]': `'ion-badge-' + label()`,
```

Comment:
Maybe you can remove this, since the element is not needed in the test cases anymore.